### PR TITLE
DAOS-15968 container: Fix CAPA fetch on NS master

### PR DESCRIPTION
**Pull request by liw**
Apparently, when an I/O request fetches a container handle on the engine who is the PS leader, if the ds_cont_hdl object is absent (because the engine has restarted), then the cont_iv code won't create a ds_cont_hdl object, even though the handle is found in the RDB. This patch adds a ds_cont_tgt_open call to cont_iv_ent_fetch, so that the flow is similar to cont_iv_ent_update.

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate owners.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

**Comment by liw** [comment]
Asking draft reviews, for I'm not familiar with the IV code and would like to know if I'm on the wrong path.

**Comment by liw** [comment]
Rebased to pick up the Ubuntu build fix.

File: src/container/container_iv.c [hunk 1]
```diff
@@ -511,8 +511,15 @@ cont_iv_ent_fetch(struct ds_iv_entry *entry, struct ds_iv_key *key,
 					}
 					prop_entry = daos_prop_entry_get(prop, DAOS_PROP_CO_STATUS);
 					D_ASSERT(prop_entry != NULL);
-
 					daos_prop_val_2_co_status(prop_entry->dpe_val, &stat);
+
+					rc = ds_cont_tgt_open(entry->ns->iv_pool_uuid,
+							      civ_key->cont_uuid, chdl.ch_cont,
+							      chdl.ch_flags, chdl.ch_sec_capas,
+							      stat.dcs_pm_ver);
```
Comment (liuxuezhao) [inline src/container/container_iv.c:519]: I just not quite understood why this change could help to resolve the problem.
Because inside cont_iv_ent_refresh(just call cont_iv_ent_update) it will call ds_cont_tgt_open(). The cont_iv_ent_refresh() suppose will be called right after cont_iv_ent_fetch() on the root node during the processing of cont_iv_fetch(..., IV_CONT_CAPA,...). 
any difference for the calling in the two places? thx

Comment (liw) [reply src/container/container_iv.c:519]: If I'm not mistaken, `ivo_on_refresh` is not called on the root node. My reading is that `crt_iv_fetch` will get 0 from `ivo_on_fetch`, and return quickly. Is that not the case?

Comment (liuxuezhao) [reply src/container/container_iv.c:519]: seems you are right, crt_hdlr_iv_fetch_aux() it will not call ivo_on_refresh() on root node.
Then it can explain why this change made difference for the test.

**Review by liuxuezhao — APPROVED** [review]

**Review by NiuYawei — APPROVED** [review]
